File: basic.js

function dealFreeCell(seed) {

  const ranks = ['A', '2', '3', '4', '5', '6', '7', '8', '9', 'T', 'J', 'Q', 'K'];

  const suits = ['C', 'D', 'H', 'S'];

  let cards = [];

  for (let rank of ranks) {

    for (let suit of suits) {

      cards.push(rank + suit);

    }

  }

  let state = seed;

  let dealOrder = [];

  for (let length = 52; length > 0; length--) {

    state = (214013 * state + 2531011) & 0x7fffffff;

    const rand = state >>> 16;

    const j = rand % length;

    // Swap

    [cards[j], cards[length - 1]] = [cards[length - 1], cards[j]];

    dealOrder.push(cards.pop());

  }

  // Group into rows for the board

  const board = [];

  for (let i = 0; i < dealOrder.length; i += 8) {

    board.push(dealOrder.slice(i, i + 8));

  }

  return board;

}

File: basic.py
def dealFreeCell(seed):
    ranks = ['A', '2', '3', '4', '5', '6', '7', '8', '9', 'T', 'J', 'Q', 'K']
    suits = ['C', 'D', 'H', 'S']
    cards = [r + s for r in ranks for s in suits]
    
    state = seed
    deal_order = []
    
    for length in range(52, 0, -1):
        state = (214013 * state + 2531011) & 0x7fffffff
        rand_val = state >> 16
        j = rand_val % length
        
        # Swap with last
        cards[j], cards[length - 1] = cards[length - 1], cards[j]
        # Take the chosen card
        deal_order.append(cards.pop())
    
    # Group into rows (as shown in the examples)
    board = []
    for i in range(0, 52, 8):
        board.append(deal_order[i:i + 8])
    
    return board
